src/crimson/osd/replicated_recovery_backend.cc

```diff
@@ -837,42 +837,20 @@ ReplicatedRecoveryBackend::_handle_pull_response(
     pull_info.recovery_info.version = push_op.version;
 
   if (pull_info.recovery_progress.first) {
-    auto obc_manager = pg.obc_loader.get_obc_manager(pull_info.recovery_info.soid);
-    co_await pg.obc_loader.load_and_lock(
-      obc_manager, RWState::RWNONE
-    ).handle_error_interruptible(
-      crimson::ct_error::assert_all("unexpected error")
-    );
+    auto [oi, ssc] = get_md_from_push_op(push_op);
```

Comment:
The issue seems to arise from here, `ssc` returned is empty.
If `(ss_attr_iter != push_op.attrset.end())` is false, we actually use the non existing `ssc` - `!pull_info.obc->ssc->exists`.
This will result in the call to `recalc_subsets` to be a no-op.

Reply:
Looking at the code again, we pass a reference `PushOp&` to `_handle_pull_response` which is a coroutine. This might be dangerous: https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#cp53-parameters-to-coroutines-should-not-be-passed-by-reference

Reply:
Fix attempt:
https://github.com/ceph/ceph-ci/commits/wip-crimson-only-recovery-69412-replica-fixes/

Reply:
There's another problem, ssc needs to be from the head, not the clone.  Working on a fix.

Reply:
On the replica side, I think the answer is probably to not populate the obc at all.  The ObjectContextLoader logic should be robust to an unloaded obc with the separation of creating a Manager from actually loading the obc -- we can lock it without populating it.